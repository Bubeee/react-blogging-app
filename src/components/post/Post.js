import React from 'react';
import './post.css';

const Post = ({ id, title, content, onChangeClick, onRemoveClick }) => {
  return (
    <div className="post-item__container">
      <h2 className="">{title}</h2>
      <p className="">{content}</p>
      <a href="#" className="btn btn-danger btn-lg remove__button">
        <span className="glyphicon glyphicon-remove-circle" onClick={() => onRemoveClick(id)}> Delete</span>
      </a>
    </div>
  );
};

export default Post;
