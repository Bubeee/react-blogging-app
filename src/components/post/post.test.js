import React from 'react';
import Post from './post';

describe('Post', () => {
  it('should render correctly', () => {
    const props = {
      id: 1,
      title: 'test',
      content: 'test post',
      onChangeClick: null,
      onRemoveClick: null
    };

    const wrapper = shallow(<Post {...props} />);

    // console.log(wrapper.getElement().props);
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.getElement().props.title.toEqual('test'));
  });
});
