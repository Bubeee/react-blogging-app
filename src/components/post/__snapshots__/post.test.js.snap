// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render correctly 1`] = `
<div
  className="post-item__container"
>
  <h2
    className=""
  >
    test
  </h2>
  <p
    className=""
  >
    test post
  </p>
  <a
    className="btn btn-danger btn-lg remove__button"
    href="#"
  >
    <span
      className="glyphicon glyphicon-remove-circle"
      onClick={[Function]}
    >
       Delete
    </span>
  </a>
</div>
`;
